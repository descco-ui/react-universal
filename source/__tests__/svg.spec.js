import React from 'react';
import { shallow } from 'enzyme';
import Svg from '../svg';

describe('Svg', function() {
  describe('Estrutura', function() {
    it('exibe o componente default', () => {
      const component = shallow(<Svg image="#icon-bank-slip" />);
      expect(component).toMatchSnapshot();
    });
  });
});
