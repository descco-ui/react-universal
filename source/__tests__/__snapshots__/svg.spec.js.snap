// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg Estrutura exibe o componente default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    image="#icon-bank-slip"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Svg
        height={22}
        onPress={undefined}
        preserveAspectRatio="xMidYMid meet"
        viewBox={undefined}
        width={36}
      >
        <G>
          <G>
            <Path
              d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
              fill="#666"
              scale="0.5"
            />
          </G>
        </G>
      </Svg>,
      "style": Object {
        "flex": 0,
        "height": 22,
        "width": 36,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <G>
          <G>
            <Path
              d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
              fill="#666"
              scale="0.5"
            />
          </G>
        </G>,
        "height": 22,
        "onPress": undefined,
        "preserveAspectRatio": "xMidYMid meet",
        "viewBox": undefined,
        "width": 36,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <G>
              <Path
                d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
                fill="#666"
                scale="0.5"
              />
            </G>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Path
                d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
                fill="#666"
                scale="0.5"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "d": "M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z",
                "fill": "#666",
                "scale": "0.5",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Svg
          height={22}
          onPress={undefined}
          preserveAspectRatio="xMidYMid meet"
          viewBox={undefined}
          width={36}
        >
          <G>
            <G>
              <Path
                d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
                fill="#666"
                scale="0.5"
              />
            </G>
          </G>
        </Svg>,
        "style": Object {
          "flex": 0,
          "height": 22,
          "width": 36,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <G>
            <G>
              <Path
                d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
                fill="#666"
                scale="0.5"
              />
            </G>
          </G>,
          "height": 22,
          "onPress": undefined,
          "preserveAspectRatio": "xMidYMid meet",
          "viewBox": undefined,
          "width": 36,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <G>
                <Path
                  d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
                  fill="#666"
                  scale="0.5"
                />
              </G>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Path
                  d="M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z"
                  fill="#666"
                  scale="0.5"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "d": "M 0 0 0 38 60 38 60 0 0 0 Z M 4 4 56 4 56 34 4 34 4 4 Z m 4 4 0 22 8 0 0 -22 -8 0 z m 10 0 0 22 6 0 0 -22 -6 0 z m 9 0 0 22 8 0 0 -22 -8 0 z m 11 0 0 22 2 0 0 -22 -2 0 z m 4 0 0 22 4 0 0 -22 -4 0 z m 6 0 0 22 4 0 0 -22 -4 0 z",
                  "fill": "#666",
                  "scale": "0.5",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            undefined,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
